pkgs/tools/nix/nixos-option/src/nixos-option.cc

```diff
@@ -583,67 +590,75 @@ void printOne(Context & ctx, Out & out, const std::string & path)
     }
 }
 
-int main(int argc, char ** argv)
-{
-    bool recursive = false;
-    std::string path = ".";
-    std::string optionsExpr = "(import <nixpkgs/nixos> {}).options";
-    std::string configExpr = "(import <nixpkgs/nixos> {}).config";
-    std::vector<std::string> args;
-
-    struct MyArgs : nix::LegacyArgs, nix::MixEvalArgs
-    {
-        using nix::LegacyArgs::LegacyArgs;
-    };
-
-    MyArgs myArgs(std::string(nix::baseNameOf(argv[0])), [&](Strings::iterator & arg, const Strings::iterator & end) {
-        if (*arg == "--help") {
-            nix::showManPage("nixos-option");
-        } else if (*arg == "--version") {
-            nix::printVersion("nixos-option");
-        } else if (*arg == "-r" || *arg == "--recursive") {
-            recursive = true;
-        } else if (*arg == "--path") {
-            path = nix::getArg(*arg, arg, end);
-        } else if (*arg == "--options_expr") {
-            optionsExpr = nix::getArg(*arg, arg, end);
-        } else if (*arg == "--config_expr") {
-            configExpr = nix::getArg(*arg, arg, end);
-        } else if (!arg->empty() && arg->at(0) == '-') {
-            return false;
-        } else {
-            args.push_back(*arg);
-        }
-        return true;
-    });
+auto main(int argc, char ** argv) -> int
+{
+    auto args = std::span(argv, argc);
+    return nix::handleExceptions(args[0], [&]() {
+        bool recursive = false;
+        std::string path = ".";
+        std::string optionsExpr = "(import <nixpkgs/nixos> {}).options";
+        std::string configExpr = "(import <nixpkgs/nixos> {}).config";
+        std::vector<std::string> args;
+
+        struct MyArgs : nix::LegacyArgs, nix::MixEvalArgs
+        {
+            MyArgs(const MyArgs& other) = default;
+            MyArgs(MyArgs&& other) noexcept = default;
+            auto operator=(const MyArgs& other) noexcept -> MyArgs& = default;
+            auto operator=(MyArgs&& other) noexcept -> MyArgs& = default;
```

Comment:
This shouldn't be necessary (I think). The virtual dtor should be defined in the base class (and I don't think nix defines it in `nix::Args` base class) if we want to destroy it via in the pointer-to-base class. Here we don't have any polymorphism.

Reply:
@xokdvium so you disagree with this lint?

```
../src/nixos-option.cc:603:16: warning: class 'MyArgs' defines a default destructor, a copy constructor, a copy assignment operator and a move constructor but does not define a mo
ve assignment operator [cppcoreguidelines-special-member-functions]
```

Reply:
Do you think the lint is useful in general?